### ISSUE TITLE
Carousel won't allow edits

### ISSUE THREAD

**Issue by trf000** [AppStateESS/carousel#9]
Upgraded to latest version of phpws, Carousel won't allow edits of any kind. Modal won't appear, checkbox for active does not set.

**Comment by jlbooker** [comment]
@Stardog Would this have to do with Bower?

@trf000, you may need to install Bower (`npm install bower`) and then run `bower install` in the main phpws directory. Bower is a front-end javascript (really Node.js/npm) package manager. The `bower install` command will go out and download external javascript packages. We'll no longer need to include jQuery in the phpws repo, nor do we need to ship it. Bower will allow you to grab the latest compatible version directly from the jQuery folks on-demand.

To confirm my theory, look at the dev console. You'll probably see 404 for a lot of js files in `phpws/bower_components/...`.

**Comment by trf000** [comment]
didn't realize I needed to install anything else. I have zero access to my
server to install this, and I'm guessing I can't just ftp it there.

**Comment by trf000** [comment]
@jlbooker @Stardog 

I'm not seeing any 404 errors regarding bower. And on a fresh installation, I was getting a ton of 500 errors regarding the settings table and a duplicate id of 0. Once I tracked down the error, I saw that i could manully change the id of each setting, but if i went into settings again and saved, it made each setting have an id of 0 again.

Again this is the default bootstrap theme. I'm not seeing a reference to bower anywhere.

**Comment by trf000** [comment]
I think the issue is in here:
```javascript
this.initEditClick = function ()
    {
        $('.pager-row').click(function (e) {
            if (!$(e.target).is('input.active-checkbox')) {
                _.initDeleteClick();
                $('#delete-slide').show();
                var row_id = $(this).data('rowId');
                $('#resource-id').val(row_id);
                $('#slide-update').modal('show');
                $('#slide-update .modal-title').html('Update slide');
                $.get('carousel/admin/', {
                    command: 'edit_slide',
                    slide_id: row_id
                }, function (data) {
                    $('#title').val(data.title);
                    $('#caption').val(data.caption);
                    $('#url').val(data.url);
                    $('#caption-zone-2').val(data.caption_zone);
                    if (data.show_title == 1) {
                        $('#show_title').prop('checked', true);
                    } else {
                        $('#show_title').prop('checked', false);
                    }
                }, 'json');
            }

        });
        $('.active-checkbox').click(function () {
            if ($(this).is(':checked')) {
                $.get('carousel/admin/', {
                    command: 'activate',
                    slide_id: $(this).data('slideId')
                });
            } else {
                $.get('carousel/admin/', {
                    command: 'deactivate',
                    slide_id: $(this).data('slideId')
                });
            }
        });
    };
```

**Comment by jlbooker** [comment]
@trf000 Are you seeing any js errors from this section in the console? It's likely that the $.get() AJAX requests are failing. There's no error handling being done on those, so if the server throws an error, the error will be quietly hidden in your browser's network request log.

Turn on the network logging, then try to edit a slide, and then have a look at the response to the AJAX GET request.

**Comment by trf000** [comment]
Weird, but I cannot see any thing like that. Every function in
slide_list.js seems to work except for the initEditClick. I'm at a complete
loss.

Can you duplicate this at all?

**Comment by Stardog** [comment]
Jeez. Sorry it took so long for me to get to this. The fix is here 562e2ddcd9908144648f2ad45f6b16fd51039467

**Comment by trf000** [comment]
Thanks Matt. I hate that I didn't figure it out on my own.
